lib/browser/api/net.ts

```diff
@@ -526,6 +529,122 @@ export function request (options: ClientRequestConstructorOptions | string, call
   return new ClientRequest(options, callback);
 }
 
+function createDeferredPromise<T, E extends Error = Error> (): { promise: Promise<T>; resolve: (x: T) => void; reject: (e: E) => void; } {
+  let res: (x: T) => void;
+  let rej: (e: E) => void;
+  const promise = new Promise<T>((resolve, reject) => {
+    res = resolve;
+    rej = reject;
+  });
+
+  return { promise, resolve: res!, reject: rej! };
+}
+
+export function fetch (input: RequestInfo, init?: RequestInit): Promise<Response> {
+  const p = createDeferredPromise<Response>();
+  let req: Request;
+  try {
+    req = new Request(input, init);
```

Comment:
Which `Request` class does this invoke ? I don't see an implementation in this file.

Reply:
IIUC, we are using the `Request` class from Node.js to process the fetch input from user and then push that result into the `Electron.net` module request class for making the actual network request thereby avoiding to duplicate the request info processing logic ?

Reply:
Yes, that's correct.